### PR TITLE
input: convert gt911 from kscan

### PR DESCRIPTION
**Pull request by fabiobaltieri**
Convert the GT911 driver to the input subsystem, fix the existing boards to work in the default config.

### CODE REVIEW

**Comment by fabiobaltieri** [comment]
Hey @danieldegrasse could you give this a test run on `mimxrt595_evk_cm33` and `mimxrt1170_evk_cm7`? It should work with no modification. Could you also check that the driver does not repeat touch-off events in polling mode? You can use `samples/subsys/input/input_dump/` for that.

Thanks!

**Comment by danieldegrasse** [comment]
Testing on the RT1170 EVK:

`CONFIG_INPUT_GT911_INTERRUPT=y`:

```
input event: dev=gt911@5d             type= 3 code=  0 value=438
input event: dev=gt911@5d             type= 3 code=  1 value=569
input event: dev=gt911@5d         SYN type= 1 code=330 value=1
input event: dev=gt911@5d             type= 3 code=  0 value=438
input event: dev=gt911@5d             type= 3 code=  1 value=569
input event: dev=gt911@5d         SYN type= 1 code=330 value=1
input event: dev=gt911@5d             type= 3 code=  0 value=438
input event: dev=gt911@5d             type= 3 code=  1 value=569
input event: dev=gt911@5d         SYN type= 1 code=330 value=1
input event: dev=gt911@5d             type= 3 code=  0 value=438
input event: dev=gt911@5d             type= 3 code=  1 value=569
input event: dev=gt911@5d         SYN type= 1 code=330 value=1
input event: dev=gt911@5d         SYN type= 1 code=330 value=0
input event: dev=gt911@5d         SYN type= 1 code=330 value=0
input event: dev=gt911@5d         SYN type= 1 code=330 value=0
```

`CONFIG_INPUT_GT911_INTERRUPT=n`:

```
input event: dev=gt911@5d             type= 3 code=  0 value=398
input event: dev=gt911@5d             type= 3 code=  1 value=631
input event: dev=gt911@5d         SYN type= 1 code=330 value=1
input event: dev=gt911@5d             type= 3 code=  0 value=398
input event: dev=gt911@5d             type= 3 code=  1 value=631
input event: dev=gt911@5d         SYN type= 1 code=330 value=1
input event: dev=gt911@5d             type= 3 code=  0 value=398
input event: dev=gt911@5d             type= 3 code=  1 value=631
input event: dev=gt911@5d         SYN type= 1 code=330 value=1
input event: dev=gt911@5d             type= 3 code=  0 value=398
input event: dev=gt911@5d             type= 3 code=  1 value=631
input event: dev=gt911@5d         SYN type= 1 code=330 value=1
input event: dev=gt911@5d         SYN type= 1 code=330 value=0
input event: dev=gt911@5d         SYN type= 1 code=330 value=0
input event: dev=gt911@5d         SYN type= 1 code=330 value=0
```

I do see three touch off events reported in both cases, but applications like LVGL seem to function fine with this PR.

**Review by danieldegrasse — DISMISSED** [review]

**Comment by fabiobaltieri** [comment]
Merge conflict, rebased, stuff moved to a shield, build tested with `$ west build -p -b mimxrt595_evk_cm33 samples/subsys/display/lvgl -DSHIELD=rk055hdmipi4m`

**Review by gmarull — DISMISSED** [review]

**Comment by danieldegrasse** [comment]
I believe the failing test here: https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/drivers/kscan/sample.yaml#L17 is caused because the KSCAN sample enables `CONFIG_KSCAN`, but not `CONFIG_INPUT`. If there is an input specific test case we could run against the GT911 driver I think it makes sense to drop this testcase, and validate the driver using the input specific one. I'll leave that decision up to you @fabiobaltieri

**Comment by fabiobaltieri** [comment]
> I believe the failing test here: https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/drivers/kscan/sample.yaml#L17 is caused because the KSCAN sample enables `CONFIG_KSCAN`, but not `CONFIG_INPUT`. If there is an input specific test case we could run against the GT911 driver I think it makes sense to drop this testcase, and validate the driver using the input specific one. I'll leave that decision up to you @fabiobaltieri

Oh yeah, the shield Kconfig has a default for it but it's in a `if LVGL` block so it did not kick in. Yeah the `gt911` driver is in the `input_buildall` test and the API have their own test so it should all be covered. I added a `CONFIG_INPUT=y` to the specific case for now but thinking more about it I think it makes more sense to drop it, let me do that then.

**Comment by danieldegrasse** [comment]
Should also drop the sample testcase: https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/drivers/kscan/sample.yaml#L17, as that is not building locally for me either.

**Comment by fabiobaltieri** [comment]
> Should also drop the sample testcase: https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/drivers/kscan/sample.yaml#L17, as that is not building locally for me either.

Right, same reason, thanks! Maybe it should have a case under `samples/subsys/display/lvgl/`? I see there's already one in `samples/drivers/display/` though so that's probably enough.

**Review by danieldegrasse — DISMISSED** [review]

**Comment by danieldegrasse** [comment]
> Right, same reason, thanks! Maybe it should have a case under samples/subsys/display/lvgl/? I see there's already one in samples/drivers/display/ though so that's probably enough.

I think a case under `samples/subsys/display/lvgl/` makes sense. I'll add that in another PR, so that we don't delay this one further.

**Comment by fabiobaltieri** [comment]
> > Right, same reason, thanks! Maybe it should have a case under samples/subsys/display/lvgl/? I see there's already one in samples/drivers/display/ though so that's probably enough.
> 
> I think a case under `samples/subsys/display/lvgl/` makes sense. I'll add that in another PR, so that we don't delay this one further.

Thanks for that. Once we'll have a LVGL native input driver it should all be less tangled. Looking forward to it but I want to finish converting all the touchscreen drivers first.

**Review by gmarull — DISMISSED** [review]

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/shields/rk055hdmipi4m/rk055hdmipi4m.overlay [hunk 1]
```diff
@@ -31,6 +31,10 @@
 		reg = <0x5d>;
 		irq-gpios = <&nxp_mipi_connector 29 GPIO_ACTIVE_HIGH>;
 		reset-gpios = <&nxp_mipi_connector 28 GPIO_ACTIVE_HIGH>;
+
+		kscan_input: kscan-input {
```
[the remark this answers is not in the document]
Comment (fabiobaltieri) [reply boards/shields/rk055hdmipi4m/rk055hdmipi4m.overlay]: I see, used `kscan_input_gt911`.

**Review by erwango — APPROVED** [review]
Approved on shields part

**Review by danieldegrasse — APPROVED** [review]

**Pull request merged**